File: CodingInterview/Programmers/Python/LV1/66_짝수와_홀수.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12937?language=python3

# Number of times solving questions: 1
# 2026-07-04

import sys, pathlib
sys.path.append(str(pathlib.Path(__file__).resolve().parent.parent))  # test_runner 경로
from test_runner import run_tests

def solution(num):
    return "Even" if num % 2 == 0 else "Odd"

# ─────────────────────────────────────────────────────────────
# 테스트
# ─────────────────────────────────────────────────────────────
test_cases = [
    (100, "Even"),
    (3, "Odd"),
]

run_tests(solution, test_cases)

File: CodingInterview/Programmers/Cpp/LV1/66_짝수와_홀수.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/12937?language=cpp
// 출처: 연습문제
// 대응 Python 풀이: ../../Python/LV1/66_짝수와_홀수.py

// Number of times solving questions: 1
// 2026-08-22

#include "../test_runner.h"
#include <string>
#include <vector>
#include <algorithm>
#include <unordered_map>
#include <unordered_set>
using namespace std;

string solution(int num) {
    string answer = "Odd";
    if (num % 2 == 0) {
        answer = "Even";
    }
    return answer;
}

int main() {
    Tester t("solution");
    t.check(solution(100), "Even");
    t.check(solution(3), "Odd");
    return t.report();
}
